src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+   * 
+   * @return std::vector<const real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<const real*> GetRedecompVelocityPtrs() const
+  {
+    return velocity_->GetRedecompFieldPtrs();
+  }
+
+  /**
+   * @brief Get the map from mesh 1 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap1() const
+  {
+    return GetUpdateData().elem_map_1_;
+  }
+
+  /**
+   * @brief Get the map from mesh 2 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap2() const
+  {
+    return GetUpdateData().elem_map_2_;
+  }
+
+  /**
+   * @brief Get the (coarse) boundary submesh containing contact surfaces
+   * 
+   * @return mfem::ParSubMesh& 
+   */
+  mfem::ParSubMesh& GetSubmesh() { return submesh_; }
+
+  /**
+   * @brief Get the (coarse) boundary submesh containing contact surfaces
+   * 
+   * @return const mfem::ParSubMesh& 
+   */
+  const mfem::ParSubMesh& GetSubmesh() const { return submesh_; }
+
+  /**
+   * @brief Get the (refined) boundary submesh containing contact surfaces
+   * 
+   * @return mfem::ParMesh*
+   */
+  mfem::ParMesh* GetLORSubmesh() { return lor_submesh_.get(); }
+
+  /**
+   * @brief Get the refined boundary submesh containing contact surfaces
+   * 
+   * @return const mfem::ParMesh*
+   *
+   * @note nullptr if no refined mesh exists
+   */
+  const mfem::ParMesh* GetLORSubmesh() const { return lor_submesh_.get(); }
+
+  /**
+   * @brief Get the (refined) redecomposed boundary submesh
+   * 
+   * @return redecomp::RedecompMesh& 
+   */
+  redecomp::RedecompMesh& GetRedecompMesh()
+  {
+    return GetUpdateData().redecomp_;
+  }
+
+  /**
+   * @brief Get the finite element space on the (coarse) boundary submesh
+   * 
+   * @return const mfem::ParFiniteElementSpace& 
+   */
+  const mfem::ParFiniteElementSpace& GetSubmeshFESpace() const
+  {
+    return *submesh_xfer_gridfn_.ParFESpace();
+  }
+
+  /**
+   * @brief Get the finite element space on the refined boundary submesh
+   * 
+   * @return const mfem::ParFiniteElementSpace* 
+   *
+   * @note nullptr if no refined mesh exists
+   */
+  const mfem::ParFiniteElementSpace* GetLORSubmeshFESpace() const
+  {
+    return lor_xfer_ ? lor_xfer_->GetLORGridFn().ParFESpace() : nullptr;
+  }
+
+  /**
+   * @brief Set the number of element subdivisions per dimension on the refined
+   * mesh
+   *
+   * @param lor_factor Number of element subdivisions per dimension
+   */
+  void SetLowOrderRefinedFactor(integer lor_factor);
+private:
+  /**
+   * @brief Creates and stores data that changes when the redecomp mesh is
+   * updated
+   */
+  struct UpdateData
+  {
+    /**
+     * @brief Construct a new UpdateData object
+     * 
+     * @param submesh Coarse boundary submesh of contact elements
```

Comment:
Consider clarity here. Maybe Parent surface submesh of contact facets? The submesh is not over the whole boundary, the submesh is not necessarily coarser than anything else if LOR isn't used, and the contact elements are one dimension lower than the volume elements.

Reply:
Fixed